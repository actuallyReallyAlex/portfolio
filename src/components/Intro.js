import React from 'react'
import PropTypes from 'prop-types'
import { Grid, Header, Button } from 'semantic-ui-react'
import SocialGroup from './SocialGroup'

export const Intro = props => {
  return (
    <section id="intro">
      <div className="intro-overlay" />
      <div className="intro-content">
        <Grid columns={1}>
          <Grid.Row>
            <Grid.Column>
              <Header as="h5" color={props.themeColor} id="header-hello-world">
                HELLO, WORLD.
              </Header>
              <Header as="h1" className="intro-header" id="header-name">
                I&#39;m {props.fullName}
              </Header>
              <p className="intro-position" id="header-position">
                <span>{props.position}</span>
              </p>
              <Button inverted color={props.themeColor}>
                More About Me
              </Button>
            </Grid.Column>
          </Grid.Row>
        </Grid>
        <SocialGroup themeColor={props.themeColor} />
      </div>
    </section>
  )
}

Intro.propTypes = {
  fullName: PropTypes.string,
  themeColor: PropTypes.string,
  position: PropTypes.string
}
