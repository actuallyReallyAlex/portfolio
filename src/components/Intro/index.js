import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading, Text } from 'grommet'
import { info, social } from '../../config.json'
import SocialIcon from '../SocialIcon'

const Intro = ({ navRef, size }) => (
  <Box
    align="center"
    background={`url(${info.heroImage})`}
    fill="horizontal"
    height={size === 'small' ? '400px' : '500px'}
    id="Home"
    justify="center"
    ref={navRef}
  >
    <Box direction="row" justify="center" align="center">
      <Box align="center" justify="center" pad="medium">
        <Text className="uppercase" color="white" margin="none" weight="bold">
          Hello, World.
        </Text>
        <Heading
          color="white"
          level="1"
          margin={{ top: 'small', bottom: 'none' }}
          size={size === 'small' ? '3em' : '4em'}
        >
          I&#39;m Alex Lee
        </Heading>
        <Text className="uppercase" color="white" margin="none" weight="bold">
          Web Developer
        </Text>
        <Box direction="row" margin={{ top: 'large' }}>
          {social.map(({ icon, link }, i) => (
            <SocialIcon
              key={i}
              icon={icon}
              link={link}
              size={size === 'small' ? '30px' : 'large'}
            />
          ))}
        </Box>
      </Box>
    </Box>
  </Box>
)

Intro.propTypes = {
  navRef: PropTypes.func.isRequired,
  size: PropTypes.string.isRequired
}

export default Intro
