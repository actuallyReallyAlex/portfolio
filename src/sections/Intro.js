import React from 'react'
import { Box, Heading } from 'grommet'
import { social } from '../config.json'
import SocialIcon from '../components/SocialIcon'

const Section = () => {
  return (
    <Box
      align="center"
      background="url(
        'https://res.cloudinary.com/alexlee-dev/image/upload/c_scale,q_90,w_1152,f_auto/v1538630398/portfolio/portfolio_portland_dark_min.jpg'
      )"
      height="400px"
      justify="center"
    >
      <Box direction="row" justify="center" align="center">
        <Box align="center" justify="center" pad="medium">
          <Heading className="uppercase" color="white" level="4" margin="none">
            Hello, World.
          </Heading>
          <Heading
            color="white"
            level="1"
            margin={{ top: 'small', bottom: 'none' }}
          >
            I&#39;m Alex Lee
          </Heading>
          <Heading className="uppercase" color="white" level="4" margin="none">
            Web Developer
          </Heading>
          <Box direction="row" margin={{ top: 'medium' }}>
            {social.map(({ icon, link }, i) => (
              <SocialIcon key={i} icon={icon} link={link} />
            ))}
          </Box>
        </Box>
      </Box>
    </Box>
  )
}

export default Section
