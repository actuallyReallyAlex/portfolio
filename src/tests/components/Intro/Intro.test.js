import React from 'react'
import Intro from '../../../components/Intro'
import { shallow } from 'enzyme'

const defaultProps = {}

describe('<Intro />', () => {
  test('Should render the <Intro />.', () => {
    const component = shallow(<Intro {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
