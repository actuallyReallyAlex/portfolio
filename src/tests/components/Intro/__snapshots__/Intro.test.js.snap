// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> Should render the <Intro />. 1`] = `ShallowWrapper {}`;
